### ISSUE TITLE
Missing feature: cut_facets

### ISSUE THREAD

**Issue by janmodderman** [gridap/STLCutters.jl#22]
Missing the cut_facets feature required to intersect non-embedded with embedded boundaries. Available in Gridap.Embedded package, should be implemented in a similar way.

**Comment by pmartorell** [comment]
Hi @janmodderman thanks for reporting.

I add @oriolcg to the conversation.

The issue is to implement: `cut_facets(::STLCutter,model::DiscreteModel,geo::STLGeometry)`

As `STLCutter` is computationally more expensive than `LevelSetCutters`. Thus, it would be convenient that 
[`cut()`](https://github.com/gridap/STLCutters.jl/blob/ad13d873cf3c01feebe90ca68fd955ec5b24b258/src/Embedded.jl#L35) returns `cutgeo,f_to_bgf,cutgeo_facets` which will avoid recomputations.

Note that `STLCutter` only supports one single geometry and cannot be intersected with a `LevelSetGeometry`.

So that, the following functions should be modified (from high to low level):

 * `cut_facets()`: new function that wraps `_cut_stl()`
 * [`cut()`](https://github.com/gridap/STLCutters.jl/blob/ad13d873cf3c01feebe90ca68fd955ec5b24b258/src/Embedded.jl#L35): return also `cutgeo_facets`
 * [`_cut_stl()`](https://github.com/gridap/STLCutters.jl/blob/ad13d873cf3c01feebe90ca68fd955ec5b24b258/src/Embedded.jl#L45): compute `EmbeddedFacetDiscretizations` data fields
 * [`subtriangulate()`](https://github.com/gridap/STLCutters.jl/blob/ad13d873cf3c01feebe90ca68fd955ec5b24b258/src/SubTriangulations.jl#L18): 
   * `_get_threaded_empty_arrays`
   *  `save_submesh!`
   *  `_append_threaded_submesh!`
 * [`save_submesh()`](https://github.com/gridap/STLCutters.jl/blob/ad13d873cf3c01feebe90ca68fd955ec5b24b258/src/SubTriangulations.jl#L180):
    * extract cell boundary simplices from polyhedra (Kin,Kout) : I can do that for you, it would look like
       ```
        T_Fin,X_Fin,f_to_lbgf = simplexify_cell_boundary(Kin)
        T_Fout,X_Fout,f_to_lbgf = simplexify_cell_boundary(Kout)
       ```
    * save simplices into mesh arrays: modifications into `_append_submesh()`

Let me know if there is anything unclear.

**Comment by janmodderman** [comment]
Hi @pmartorell, 

I looked into adding this feature this week and I have a question on the last 2 functions; subtriangulate() and save_submesh().

In the save_submesh() function calling the simplexify on the inner and outer boundary is straightforward, I was wondering how you to save the simplices into the mesh arrays. I assume that _append_subfacets should be used for this modification, but should that replace the current subfacets being appended as well? Or just have 2 additional append subfacets calls?

Secondly, should all the functions placed under the subtriangulate() bullet be modified or only the save_submesh() function?

Best regards,

Jan

**Comment by janmodderman** [comment]
Hi @pmartorell, 

Thank you for the quick reply! I made a fork which is available here https://github.com/janmodderman/STLCutters.jl 

I pushed the changes I have made up to this point to this fork. 

Best regards,

Jan Modderman

**Comment by pmartorell** [comment]
After a meeting with @janmodderman on 14 Dec 2023, the workflow is functioning (https://github.com/janmodderman/STLCutters.jl/commit/9ef7ee63b5ce49000c79efeca8810a4f4f25ada9)

Still, there are the following tasks to do:

 * [ ] Code cleaning and style (@pmartorell)
    * [ ] Merge arrays
    * [ ] Remove inline annotations
 * [ ] Missing devs (@pmartorell)
    * [ ] Remove small facets
    * [ ] Append threaded arrays
    * [ ] Extend `AbstractEmbeddedDiscretization` to hold both
 * [ ] Testing (@janmodderman)
    * [ ] Test area vs. known value
    * [ ] Test area in + out vs. total
    * [ ] Moment fitting test (@pmartorell)
    * [ ] Functional example

**Comment by pmartorell** [comment]
This feature will be available in the next release (v0.2.0)
